lib/basedriver/commands/find.js

```diff
@@ -341,35 +359,74 @@ helpers.getScreenshotForImageFind = async function (screenWidth, screenHeight) {
   // are two potential types of mismatch: aspect ratio mismatch and scale
   // mismatch. we need to detect and fix both
 
-  const screenAR = screenWidth / screenHeight;
-  const shotAR = shotWidth / shotHeight;
-
-  if (screenAR === shotAR) {
-    log.info('Screenshot aspect ratio matched screen aspect ratio');
-  } else {
-    log.warn(`When trying to find an element, determined that the screen ` +
-             `aspect ratio and screenshot aspect ratio are different. Screen ` +
-             `is ${screenWidth}x${screenHeight} whereas screenshot is ` +
-             `${shotWidth}x${shotHeight}.`);
-    shotWidth = shotWidth / (shotAR / screenAR);
-    log.warn(`Resizing screenshot to ${shotWidth}x${shotHeight} to match ` +
-             `screen aspect ratio so that image element coordinates have a ` +
-             `greater chance of being correct.`);
-    imgObj = imgObj.resize(shotWidth, shotHeight);
-  }
-
-  // now we know the aspect ratios match, but there might still be a scale
-  // mismatch, so just resize based on the screen dimensions
-  if (screenWidth !== shotWidth) {
+  // const screenAR = screenWidth / screenHeight;
+  // const shotAR = shotWidth / shotHeight;
+
+  // if (screenAR === shotAR) {
+  //   log.info('Screenshot aspect ratio matched screen aspect ratio');
+  // } else {
+  //   log.warn(`When trying to find an element, determined that the screen ` +
+  //            `aspect ratio and screenshot aspect ratio are different. Screen ` +
+  //            `is ${screenWidth}x${screenHeight} whereas screenshot is ` +
+  //            `${shotWidth}x${shotHeight}.`);
+  //   shotWidth = shotWidth / (shotAR / screenAR);
+  //   log.warn(`Resizing screenshot to ${shotWidth}x${shotHeight} to match ` +
+  //            `screen aspect ratio so that image element coordinates have a ` +
+  //            `greater chance of being correct.`);
+  //   imgObj = imgObj.resize(shotWidth, shotHeight);
+  // }
+
+  // Resize based on the screen dimensions only if both width and height are mismatched
+  // since except for that, it might be a situation which is different window rect and
+  // screenshot size like `@driver.window_rect #=>x=0, y=0, width=1080, height=1794` and
+  // `"deviceScreenSize"=>"1080x1920"`
+  let scale;
+  if (screenWidth !== shotWidth && screenHeight !== shotHeight) {
     log.info(`Scaling screenshot from ${shotWidth}x${shotHeight} to match ` +
              `screen at ${screenWidth}x${screenHeight}`);
     imgObj = imgObj.resize(screenWidth, screenHeight);
+
+    scale = {widthRatio: screenWidth / shotWidth, heightRatio: screenHeight / shotHeight};
```

Comment:
this might be an integer division. how about adding `1.0 *` to each expression?

Reply:
mm. If the result is a float, then will this float?

```node
> 1098 / 500
2.196
```

I quick checked. `1.0` work as an integer in JS??

```node
> function isFloat(n){
...     return Number(n) === n && n % 1 !== 0;
... }
undefined
> a = 1.1
1.1
> isFloat(a)
true
> a = 1.0
1
> isFloat(a)
false
> parseFloat(a)
1
> isFloat(parseFloat(a))
false
```

Reply:
yeah, I'm constantly forgetting that Javascript has many unordinary things like the missing integer division concept, for example :)

Reply:
ah.. gotha :)